src/testing/code-debugging.md

```diff
@@ -675,6 +675,21 @@ The `RepaintBoundary` widget, which creates a `RenderRepaintBoundary`
 in the render tree, creates a new layer in the layer tree. This is
 used to reduce how much needs to be repainted.
 
+### Focus tree
+
+To debug a focus or shortcut issue, you can use dump the focus tree
+using [`debugDumpFocusTree()`][].
+
+For example:
+
+```
+I/flutter : FocusManager#03d71
+I/flutter : └─rootScope: FocusScopeNode#eefc1(Root Focus Scope)
```

Comment:
Maybe a slightly larger example?  Maybe just add a `MaterialApp()`, which would add another focus scope, and a button? I'm just thinking that the root focus scope is never really accessed, so people wouldn't know what that was for.

Reply:
Does that sample scroll in the actual page?  I'm hoping it will be a fixed size so it doesn't take up so much space on the page, but be scrollable so they can see the whole thing if they want. I think it's good to have the real world example, but it is kind of long... :-)

Reply:
Yup it's definitely long and sadly it's not a fixed size. The render tree dump section has the example same problem: https://docs.flutter.dev/testing/code-debugging#render-tree

Reply:
@sfshaza2 Is there another kind of code section/HTML we can use to make the box be limited size but scrollable? I'm thinking of what GitHub does with code sections, like so:

```
I/flutter : FocusManager#6fb59
I/flutter :  │ primaryFocus: FocusScopeNode#3c26f(_ModalScopeState<dynamic>
I/flutter :  │   Focus Scope [PRIMARY FOCUS])
I/flutter :  │ primaryFocusCreator: FocusScope ← PrimaryScrollController ←
I/flutter :  │   _ActionsScope ← Actions ← Builder ← PageStorage ← Offstage ←
I/flutter :  │   _ModalScopeStatus ← UnmanagedRestorationScope ←
I/flutter :  │   RestorationScope ← AnimatedBuilder ←
I/flutter :  │   _ModalScope<dynamic>-[LabeledGlobalKey<_ModalScopeState<dynamic>>#f36a2]
I/flutter :  │   ← Semantics ← _RenderTheaterMarker ← _EffectiveTickerMode ←
I/flutter :  │   TickerMode ←
I/flutter :  │   _OverlayEntryWidget-[LabeledGlobalKey<_OverlayEntryWidgetState>#2e2a3]
I/flutter :  │   ← _Theater ← Overlay-[LabeledGlobalKey<OverlayState>#89fc1] ←
I/flutter :  │   UnmanagedRestorationScope ← ⋯
I/flutter :  │
I/flutter :  └─rootScope: FocusScopeNode#95ff1(Root Focus Scope [IN FOCUS PATH])
I/flutter :    │ IN FOCUS PATH
I/flutter :    │ focusedChildren: FocusScopeNode#001cc(Navigator Scope [IN FOCUS
I/flutter :    │   PATH])
I/flutter :    │
I/flutter :    └─Child 1: FocusNode#79786([IN FOCUS PATH])
I/flutter :      │ context: Focus
I/flutter :      │ NOT FOCUSABLE
I/flutter :      │ IN FOCUS PATH
I/flutter :      │
I/flutter :      └─Child 1: FocusNode#15aec(Shortcuts [IN FOCUS PATH])
I/flutter :        │ context: Focus
I/flutter :        │ NOT FOCUSABLE
I/flutter :        │ IN FOCUS PATH
I/flutter :        │
I/flutter :        └─Child 1: FocusNode#3514b(Shortcuts [IN FOCUS PATH])
I/flutter :          │ context: Focus
I/flutter :          │ NOT FOCUSABLE
I/flutter :          │ IN FOCUS PATH
I/flutter :          │
I/flutter :          └─Child 1: _FocusTraversalGroupNode#0ccda(FocusTraversalGroup [IN FOCUS PATH])
I/flutter :            │ context: Focus
I/flutter :            │ NOT FOCUSABLE
I/flutter :            │ IN FOCUS PATH
I/flutter :            │
I/flutter :            └─Child 1: FocusNode#e2413(Shortcuts [IN FOCUS PATH])
I/flutter :              │ context: Focus
I/flutter :              │ NOT FOCUSABLE
I/flutter :              │ IN FOCUS PATH
I/flutter :              │
I/flutter :              └─Child 1: FocusScopeNode#001cc(Navigator Scope [IN FOCUS PATH])
I/flutter :                │ context: FocusScope
I/flutter :                │ IN FOCUS PATH
I/flutter :                │ focusedChildren: FocusScopeNode#3c26f(_ModalScopeState<dynamic>
I/flutter :                │   Focus Scope [PRIMARY FOCUS])
I/flutter :                │
I/flutter :                └─Child 1: _FocusTraversalGroupNode#1d456(FocusTraversalGroup [IN FOCUS PATH])
I/flutter :                  │ context: Focus
I/flutter :                  │ NOT FOCUSABLE
I/flutter :                  │ IN FOCUS PATH
I/flutter :                  │
I/flutter :                  └─Child 1: FocusNode#3635f(Navigator [IN FOCUS PATH])
I/flutter :                    │ context: Focus
I/flutter :                    │ IN FOCUS PATH
I/flutter :                    │
I/flutter :                    └─Child 1: FocusScopeNode#3c26f(_ModalScopeState<dynamic> Focus Scope [PRIMARY FOCUS])
I/flutter :                        context: FocusScope
I/flutter :                        PRIMARY FOCUS
```